File: src/RealWorld.Core/Entities/Comment.cs
namespace RealWorld.Core.Entities;

public class Comment
{
    public string Id { get; set; } = string.Empty;
    public string Body { get; set; } = string.Empty;
    public string UserId { get; set; } = string.Empty;
    public string ArticleId { get; set; } = string.Empty;
    public DateTimeOffset CreatedAt { get; set; }

    public Comment() { }

    public Comment(string body, string userId, string articleId)
    {
        Id = Guid.NewGuid().ToString();
        Body = body;
        UserId = userId;
        ArticleId = articleId;
        CreatedAt = DateTimeOffset.UtcNow;
    }

    public override bool Equals(object? obj) => obj is Comment c && c.Id == Id;
    public override int GetHashCode() => Id.GetHashCode();
}

File: comments-service/src/main/java/io/spring/comments/core/comment/Comment.java
package io.spring.comments.core.comment;

import java.util.UUID;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.joda.time.DateTime;

@Getter
@NoArgsConstructor
@EqualsAndHashCode(of = "id")
public class Comment {
  private String id;
  private String body;
  private String userId;
  private String articleId;
  private DateTime createdAt;

  public Comment(String body, String userId, String articleId) {
    this(UUID.randomUUID().toString(), body, userId, articleId, new DateTime());
  }

  public Comment(String id, String body, String userId, String articleId, DateTime createdAt) {
    this.id = id == null ? UUID.randomUUID().toString() : id;
    this.body = body;
    this.userId = userId;
    this.articleId = articleId;
    this.createdAt = createdAt == null ? new DateTime() : createdAt;
  }
}
